### PR TITLE
CRUD user

### PR DESCRIPTION
**Pull request by nhuquynh2603**
Create 5 api
- Get list user: /api/user/
exp: ttp://localhost:3000/api/users?keySearch=quynh123&&page=0&limit=5
- Get single user: GET /api/user/{id}  
exp: http://localhost:3000/api/users/1
- Delete user: DELETE  /api/user/{id}   
exp: http://localhost:3000/api/users/1
- Update user: PUT /api/user/{id}
exp: http://localhost:3000/api/users/1
- Create user: /api/user/
exp: http://localhost:3000/api/users/

### CODE REVIEW

File: project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.support.ResourceBundleMessageSource;
 
 @Configuration
 public class MessageSourceConfig {
+
+    @Bean
+    public Constants constants() {
+        return new Constants();
+    }
```
Comment (phanquang-dtvn) [inline project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java:15]: Tại sao tạo bean Constants hè?

File: project/src/main/java/com/example/project/service/ErrorService.java
Comment (phanquang-dtvn) [inline project/src/main/java/com/example/project/service/ErrorService.java:1]: Em tạo Service message là gì vậy? Hơi bị phức tạp code rồi.

**Review by phanquang-dtvn — COMMENTED** [review]
@nhuquynh2603

File: project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java (hunk 1 above)
Comment (nhuquynh2603) [reply project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java:15]: @phanquang-dtvn e dùng để inject Constants vào  UserValidator với User Controller.

File: project/src/main/java/com/example/project/service/ErrorService.java
Comment (nhuquynh2603) [reply project/src/main/java/com/example/project/service/ErrorService.java:1]: @phanquang-dtvn dạ e sửa lại rồi ạ
https://github.com/dtvn-training/2023-dn-java1-project-backend/pull/2/commits/e136b20cc8cbe238fc921ea2b591bdd87e65b642

File: project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java (hunk 1 above)
Comment (nhuquynh2603) [reply project/src/main/java/com/example/project/configuration/custom/MessageSourceConfig.java]: @phanquang-dtvn  dạ e đã sửa lại ạ
https://github.com/dtvn-training/2023-dn-java1-project-backend/pull/2/commits/8dde84ce049a07b4af94020e63246d2129f2ad2d

**Pull request merged**